pkg/services/ngalert/writer/prom.go

```diff
@@ -66,9 +66,15 @@ func PointsFromFrames(name string, t time.Time, frames data.Frames, extraLabels
 
 	points := make([]Point, 0, len(col.Refs))
 	for _, ref := range col.Refs {
-		fp, empty, _ := ref.NullableFloat64Value()
-		if empty || fp == nil {
-			return nil, fmt.Errorf("unable to read float64 value")
+		fp, empty, err := ref.NullableFloat64Value()
+		if empty {
+			return nil, fmt.Errorf("empty frame")
+		}
+		if fp == nil {
+			return nil, fmt.Errorf("nil frame")
+		}
+		if err != nil {
```

Comment:
Just came to mind that we should actually check `err` before `fp == nil`, otherwise we might still suppress errors